interoperator/controllers/provisioners/sfclusterusage/sfclusterusage_controller.go

```diff
@@ -0,0 +1,168 @@
+/*
+Copyright 2019 The Service Fabrik Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sfclusterusage
+
+import (
+	"context"
+
+	resourcev1alpha1 "github.com/cloudfoundry-incubator/service-fabrik-broker/interoperator/api/resource/v1alpha1"
+	"github.com/cloudfoundry-incubator/service-fabrik-broker/interoperator/pkg/constants"
+	"github.com/cloudfoundry-incubator/service-fabrik-broker/interoperator/pkg/watches"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/retry"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+// SFClusterUsageReconciler reconciles a Node objects and computes the capacity of cluster
+type SFClusterUsageReconciler struct {
+	client.Client
+	Log    logr.Logger
+	Scheme *runtime.Scheme
+}
+
+// Reconcile iterates through all nodes and computes requested resources
+// and update it in sfcluster status
+func (r *SFClusterUsageReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	ctx := context.Background()
+	log := r.Log.WithValues("SFCluster", req.NamespacedName)
+
+	cluster := &resourcev1alpha1.SFCluster{}
+	err := r.Get(ctx, req.NamespacedName, cluster)
+	if err != nil {
+		if apiErrors.IsNotFound(err) {
+			// Object not found, return.
+			return ctrl.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return ctrl.Result{}, err
+	}
+
+	clusterID := cluster.GetName()
+	if clusterID != constants.OwnClusterID {
+		// Should compute only for own cluster
+		return ctrl.Result{}, nil
+	}
+
+	nodes := &corev1.NodeList{}
+	currentCapacity := make(corev1.ResourceList)
+	// TODO: Verify pagination works
```

Comment:
Is this verified already ? The code seems to be in place for pagination.

Reply:
I never tested this with large number of nodes/pods. That is why I put a comment. Will remove the comment.

Reply:
Let us have it then.